File: examples/kotlin/src/test/kotlin/FirstScriptTest.kt
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.openqa.selenium.By
import org.openqa.selenium.WebDriver
import org.openqa.selenium.chrome.ChromeDriver
import java.time.Duration


class FirstScriptTest {

    @Test
    fun eightComponents() {
        val driver: WebDriver?
        driver = ChromeDriver()
        driver.get("https://google.com")
        Assertions.assertEquals("Google", driver.getTitle())
        driver.manage().timeouts().implicitlyWait(Duration.ofMillis(500))
        var searchBox = driver.findElement(By.name("q"))
        val searchButton = driver.findElement(By.name("btnK"))
        searchBox.sendKeys("Selenium")
        searchButton.click()
        searchBox = driver.findElement(By.name("q"))
        Assertions.assertEquals("Selenium", searchBox.getAttribute("value"))
        driver.quit()
    }
}

File: examples/scala/src/test/scala/FirstScriptTest.scala
import io.github.bonigarcia.wdm.WebDriverManager
import org.openqa.selenium.chrome.ChromeDriver
import org.scalatest.BeforeAndAfterAll
import org.scalatest.concurrent.Eventually
import org.scalatest.flatspec.AnyFlatSpec
import org.scalatest.matchers.should.Matchers
import org.scalatest.time.{Milliseconds, Span}
import org.scalatestplus.selenium.WebBrowser

class FirstScriptTest extends AnyFlatSpec with BeforeAndAfterAll with Matchers with Eventually with WebBrowser {
  implicit val webDriver: ChromeDriver = {
    WebDriverManager.chromedriver().setup()
    new ChromeDriver()
  }

  it should "be able to search on google" in {
    go to "https://google.com"
    pageTitle should be("Google")
    implicitlyWait(Span(500, Milliseconds))
    textField("q").value = "Selenium"
    click on name("btnK")
    eventually { textField("q").value should be ("Selenium") }
  }

  override def afterAll(): Unit = {
    quit()
  }

}
